# References and Citations

While not all of the listed literature and software were used in our final experimental approach and/or bioinformatic pipeline, they all helped to form our understanding of cfDNA, variant calling, and all the intricate steps between these two. 

## Software Used:
---
### 1. lofreq (Variant Calling)

Literature:

- LoFreq: a sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets, (available at https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3526318/).

GitHub Repo:

- https://csb5.github.io/lofreq/

### 2. bwa mem (Burrows-Wheeler Alignment)

Manual:

- http://bio-bwa.sourceforge.net/bwa.shtml

GitHub Repo:

- https://github.com/lh3/bwa

## Literature Consulted
---

### 1. eVIDENCE:

- K. Mizuno, S. Akamatsu, T. Sumiyoshi, J. H. Wong, M. Fujita, K. Maejima, K. Nakano, A. Ono, H. Aikata, M. Ueno, S. Hayami, H. Yamaue, K. Chayama, T. Inoue, O. Ogawa, H. Nakagawa, A. Fujimoto, eVIDENCE: a practical variant filtering for low-frequency variants detection in cell-free DNA. Sci Rep. 9, 15017 (2019).

### 2. Sample Prep Methods Background

- C. K. Raymond, F. C. Raymond, K. Hill, UltraPrep is a scalable, cost-effective, bead-based method for purifying cell-free DNA. PLOS ONE. 15, e0231854 (2020).

### 3. cfDNA Background

- A. Kustanovich, R. Schwartz, T. Peretz, A. Grinshpun, Life and death of circulating cell-free DNA. Cancer Biol Ther. 20, 1057â€“1067 (2019).

