# References and Citations

**NOTE:** 

While not all of the listed literature and software were used in our final experimental approach and/or bioinformatic pipeline, they all helped to form our understanding of cfDNA, variant calling, and the intricate steps between these two. 

## Software Used:
---
### 1. bwa (Burrows-Wheeler Alignment)

Manual:

- http://bio-bwa.sourceforge.net/bwa.shtml

GitHub Repo:

- https://github.com/lh3/bwa

### 2. lofreq (Variant Calling)

Literature:

- LoFreq: a sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets, (available at https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3526318/).

GitHub Repo:

- https://csb5.github.io/lofreq/

### 3. FastQC

Babraham Institute Manual and Information:

- https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

### 4. cutadapt

GitHub Repo: 

- https://github.com/marcelm/cutadapt

## Literature Consulted

### 1. Sample Prep Methods Background

- C. K. Raymond, F. C. Raymond, K. Hill, UltraPrep is a scalable, cost-effective, bead-based method for purifying cell-free DNA. PLOS ONE. 15, e0231854 (2020).

### 2. cfDNA and Liquid Biopsy Background

- A. Kustanovich, R. Schwartz, T. Peretz, A. Grinshpun, Life and death of circulating cell-free DNA. Cancer Biol Ther. 20, 1057–1067 (2019).

- S. Mohan, M. Ayub, D. G. Rothwell, S. Gulati, B. Kilerci, A. Hollebecque, H. Sun Leong, N. K. Smith, S. Sahoo, T. Descamps, C. Zhou, R. A. Hubner, M. G. McNamara, A. Lamarca, J. W. Valle, C. Dive, G. Brady, Analysis of circulating cell-free DNA identifies KRAS copy number gain and mutation as a novel prognostic marker in Pancreatic cancer. Sci Rep. 9, 11610 (2019).

- P. Razavi, B. T. Li, D. N. Brown, B. Jung, E. Hubbell, R. Shen, W. Abida, K. Juluru, I. De Bruijn, C. Hou, O. Venn, R. Lim, A. Anand, T. Maddala, S. Gnerre, R. Vijaya Satya, Q. Liu, L. Shen, N. Eattock, J. Yue, A. W. Blocker, M. Lee, A. Sehnert, H. Xu, M. P. Hall, A. Santiago-Zayas, W. F. Novotny, J. M. Isbell, V. W. Rusch, G. Plitas, A. S. Heerdt, M. Ladanyi, D. M. Hyman, D. R. Jones, M. Morrow, G. J. Riely, H. I. Scher, C. M. Rudin, M. E. Robson, L. A. Diaz, D. B. Solit, A. M. Aravanis, J. S. Reis-Filho, High-intensity sequencing reveals the sources of plasma circulating cell-free DNA variants. Nat Med. 25, 1928–1937 (2019).0

- C. D. M. Campos, K. Childers, S. S. T. Gamage, H. Wijerathne, Z. Zhao, S. A. Soper, Analytical Technologies for Liquid Biopsy of Subcellular Materials. Annu Rev Anal Chem (Palo Alto Calif). 14, 207–229 (2021).

- Z. Pös, O. Pös, J. Styk, A. Mocova, L. Strieskova, J. Budis, L. Kadasi, J. Radvanszky, T. Szemes, Technical and Methodological Aspects of Cell-Free Nucleic Acids Analyzes. Int J Mol Sci. 21, 8634 (2020).

- S. Jenkins, J. C.-H. Yang, S. S. Ramalingam, K. Yu, S. Patel, S. Weston, R. Hodge, M. Cantarini, P. A. Jänne, T. Mitsudomi, G. D. Goss, Plasma ctDNA Analysis for Detection of the EGFR T790M Mutation in Patients with Advanced Non–Small Cell Lung Cancer. Journal of Thoracic Oncology. 12, 1061–1070 (2017).


